src/mailadm/bot.py

```diff
@@ -43,119 +43,70 @@ def __init__(self, db: DB, account: deltachat.Account):
 
     @account_hookimpl
     def ac_incoming_message(self, message: deltachat.Message):
-        arguments = message.text.split(" ")
-        logging.info("process_incoming_message: %s", message.text)
-        if not self.is_admin_group_message(message):
+        """This method is called on every incoming message and decides what to do with it."""
+        logging.info("new message from %s: %s" % (message.get_sender_contact().addr, message.text))
+        if self.is_admin_group_message(message):
+            if message.text[0] == "/":
+                logging.info("%s seems to be a valid command.", message.text)
+                self.handle_command(message)
+            else:
+                logging.info("ignoring message, it's just admins discussing in the admin group")
+        elif self.is_support_group(message.chat):
+            if message.quote:
+                if message.quote.get_sender_contact().addr == self.account.get_config("addr"):
+                    self.forward_reply_to_support_user(message)
+            elif message.text[0] == "/":
+                logging.info("ignoring command, it wasn't given in the admin group")
+                message.chat.send_text("Sorry, I only take commands in the admin group.")
+            else:
+                logging.info("ignoring message, it's just admins discussing in a support group")
+        else:
             chat = message.create_chat()
             if chat.is_group():
-                if message.get_sender_contact() not in self.admingroup.get_contacts():
-                    logging.info("%s added me to a group, I'm leaving it.",
-                                 message.get_sender_contact().addr)
-                    chat.send_text("Sorry, you can not contact me in a group chat. Please use a 1:1"
-                                   " chat.")
-                    chat.remove_contact(self.account.get_self_contact())   # leave group
-                elif message.quote:  # reply to user
-                    if message.quote.get_sender_contact().addr == self.account.get_config("addr"):
-                        recipient = message.quote.override_sender_name
-                        logging.info("I'm forwarding the admin reply to the support user %s.",
-                                     recipient)
-                        chat = self.account.create_chat(recipient)
-                        chat.send_msg(message)
-                else:
-                    logging.info("ignoring message, it's just admins discussing in a support group")
+                logging.info("%s added me to a group, I'm leaving it.",
+                             message.get_sender_contact().addr)
+                chat.send_text("Sorry, you can not contact me in groups. Please use a 1:1 chat.")
+                chat.remove_contact(self.account.get_self_contact())   # leave group
+            elif message.text[0:5] == "/help":
```

Comment:
> This also reacts to `/helpfoobar`, probably not a big deal.

that's intended, as it also catches typos like `/genqr` instead of `/gen-qr` ;) then a help message helps the admin on the second try